env/env.cc

```diff
@@ -693,6 +696,81 @@ Status Env::GetHostNameString(std::string* result) {
   return s;
 }
 
+namespace {
+
+// For GenerateRawUuid()
+struct RawUuidHelper {
+  // All the data we want to go into the initial random seed. For safety in
```

Comment:
This is not really related to the UUID but is more a random number generator.  Can we somehow unify this with the Random class and create one implementation of Random (or make it a virtual method off Env)?